src/libraries/System.Text.RegularExpressions/tests/System.Text.RegularExpressions.Generators.Tests/RegexGeneratorParserTests.cs

```diff
@@ -442,6 +442,30 @@ partial class C
             ", compile: true));
         }
 
+        [Fact]
+        public async Task Valid_ClassWithGenericConstraints()
+        {
+            Assert.Empty(await RunGenerator(@"
+                using D;
+                using System.Text.RegularExpressions;
+                namespace A
+                {
+                    public partial class B<U>
+                    {
+                        private partial class C<T> where T : IBlah
+                        {
+                            [RegexGenerator(""ab"")]
+                            private static partial Regex Valid();
```

Comment:
It might be good to also include a partial method that accepts the generic type.  That way you ensure that the generator can still handle / refer to the generic parameters.

IOW: `Valid<T>(T value)`

Reply:
The partial method that's implemented by the generator is prohibited from being generic:
https://github.com/dotnet/runtime/blob/64f48857a6193a81faa32825cc78d7340a3ebd46/src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Parser.cs#L129

Reply:
I see and parameterless.  Fair enough.  This might be a more interesting scenario to test in the logging generator.